lib/api/controllers/bulkController.js

```diff
@@ -55,6 +66,63 @@ class BulkController extends BaseController {
         return response;
       });
   }
+
+  /**
+   * Write a document without adding metadata or performing data validation.
+   * @param {Request} request
+   * @returns {Promise<Object>}
+   */
+  write(request) {
+    assertHasBody(request);
+    assertHasIndexAndCollection(request);
+    assertIdStartsNotUnderscore(request);
+
+    const notify = this.ensureBooleanFlag(request, 'input.args.notify');
+
+    return this.engine.writeDocument(request)
+      .then(response => {
+        this.kuzzle.indexCache.add(request.input.resource.index, request.input.resource.collection);
+
+
+        if (notify && response.created) {
+          this.kuzzle.notifier.notifyDocumentCreate(request, response);
+        }
+        else if (notify) {
```

Comment:
Isn't this condition redundant ?

Reply:
No because I want to notify only if the user specified that he want realtime notifications

Reply:
see on slack, it's redundancy vs imbrication